src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3346,6 +3358,176 @@ public AttributeProviderPair (CustomAttribute attribute, ICustomAttributeProvide
+								methodReturnValue = methodParams [0];
+							}
+							break;
+
+						//
+						// static CreateInstance (System.Type type)
+						// static CreateInstance (System.Type type, bool nonPublic)
+						// static CreateInstance (System.Type type, params object?[]? args)
+						// static CreateInstance (System.Type type, object?[]? args, object?[]? activationAttributes)
+						// static CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object?[]? args, System.Globalization.CultureInfo? culture)
+						// static CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object?[]? args, System.Globalization.CultureInfo? culture, object?[]? activationAttributes) { throw null; }
+						//
+						case "CreateInstance" when !calledMethod.ContainsGenericParameter
+							&& calledMethod.DeclaringType.Name == "Activator"
+							&& calledMethod.Parameters.Count >= 1
+							&& calledMethod.Parameters[0].ParameterType.MetadataType != MetadataType.String: {
+
+								var parameters = calledMethod.Parameters;
+
+								reflectionContext.AnalyzingPattern ();
+
+								int? ctorParameterCount = null;
+								BindingFlags bindingFlags = BindingFlags.Instance;
+								if (parameters.Count > 1) {									
+									if (parameters [1].ParameterType.MetadataType == MetadataType.Boolean) {
+										// The overload that takes a "nonPublic" bool
+										bool nonPublic = true;
+										if (methodParams [1] is ConstIntValue constInt) {
+											nonPublic = constInt.Value != 0;
+										}
+
+										if (nonPublic)
+											bindingFlags |= BindingFlags.NonPublic | BindingFlags.Public;
+										else
+											bindingFlags |= BindingFlags.Public;
+										ctorParameterCount = 0;
+									} else {
+										// Overload that has the parameters as the second or fourth argument
+										int argsParam = parameters.Count == 2 || parameters.Count == 3 ? 1 : 3;
+										
+										if (methodParams.Count > argsParam &&
+											methodParams [argsParam] is ArrayValue arrayValue &&
+											arrayValue.Size.AsConstInt () != null) {
+											ctorParameterCount = arrayValue.Size.AsConstInt ();
+										}
+
+										if (parameters.Count > 3) {
+											if (methodParams [1].AsConstInt () != null)
+												bindingFlags |= (BindingFlags)methodParams [1].AsConstInt ();
+											else
+												bindingFlags |= BindingFlags.NonPublic | BindingFlags.Public;
+										} else {
+											bindingFlags |= BindingFlags.Public;
+										}
+									}
+								}
+								else {
+									// The overload with a single System.Type argument
+									ctorParameterCount = 0;
+									bindingFlags |= BindingFlags.Public;
+								}
+
+								// Go over all types we've seen
+								foreach (var value in methodParams[0].UniqueValues ()) {
+									if (value is SystemTypeValue systemTypeValue) {
```

Comment:
We'll probably have to change the reporting here
The context expects to get basically one call per analysis - either recognized or not. Since this is a foreach, it may get multiple. It probably just needs reviewing.